### PR TITLE
Fixing memory leaks

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
|[Issue 675](https://github.com/wazuh/wazuh-qa/issues/675)|

## Description

This PR fixes some memory leaks discovered during the performance analysis of the overall implementation of the NVD support for Linux agents in Vulnerability Detector.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [x] Source installation
- [ ] Package installation
- [x] Source upgrade
- [ ] Package upgrade

### CODE REVIEW

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -1007,7 +1007,11 @@ int wm_vuldet_fill_report_oval_data(sqlite3 *db, sqlite3_stmt *stmt, agent_softw
         sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
         sqlite3_bind_text(stmt, 2, target, -1, NULL);
 
-        ref_count = 0;
+        for (ref_count = 0; report->references && report->references[ref_count]; ++ref_count) {
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1010]: Considering that in this part of the code, we are adding some extra information to the `report` structure apart from the NVD information that was already added, we need to check whether that and if so, count the number of references in order to append the new ones.
If we don't count the references, we will not be appending information to the array and as consequence, we will lose the memory references of the information already added, causing in this way a memory leak.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -1038,6 +1042,11 @@ int wm_vuldet_fill_report_oval_data(sqlite3 *db, sqlite3_stmt *stmt, agent_softw
         sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
         sqlite3_bind_text(stmt, 2, target, -1, NULL);
 
+        for (bugref_count = 0; report->bugzilla_references && report->bugzilla_references[bugref_count]; ++bugref_count) {
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1045]: DITTO

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 3]
```diff
@@ -1055,6 +1064,11 @@ int wm_vuldet_fill_report_oval_data(sqlite3 *db, sqlite3_stmt *stmt, agent_softw
         sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
         sqlite3_bind_text(stmt, 2, target, -1, NULL);
 
+        for (adv_count = 0; report->advisories && report->advisories[adv_count]; ++adv_count) {
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1067]: DITTO

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 4]
```diff
@@ -2215,9 +2229,10 @@ int wm_vuldet_insert(wm_vuldet_db *parsed_oval, update_node *update) {
                 return wm_vuldet_sql_error(db, stmt);
             }
             wdb_finalize(stmt);
-            free(info_it->refs->values[j]);
         }
 
+        free_strarray(info_it->refs->values);
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:2234]: We were only freeing the memory of each value in the array but not the array itself.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 5]
```diff
@@ -2230,9 +2245,10 @@ int wm_vuldet_insert(wm_vuldet_db *parsed_oval, update_node *update) {
                 return wm_vuldet_sql_error(db, stmt);
             }
             wdb_finalize(stmt);
-            free(info_it->bugzilla_references->values[j]);
         }
 
+        free_strarray(info_it->bugzilla_references->values);
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:2250]: DITTO

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 6]
```diff
@@ -2245,9 +2261,10 @@ int wm_vuldet_insert(wm_vuldet_db *parsed_oval, update_node *update) {
                 return wm_vuldet_sql_error(db, stmt);
             }
             wdb_finalize(stmt);
-            free(info_it->advisories->values[j]);
         }
 
+        free_strarray(info_it->advisories->values);
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:2266]: DITTO

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** f950cdc8e2bd2903ef5edb4873cce78e9d0c157d
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1959/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1959/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1959/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1959.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1959/Analysis_dynamic_valgrind-v3.13.0-B1959.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** f950cdc8e2bd2903ef5edb4873cce78e9d0c157d
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1960/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1960/ossec.log)
**Coverage:** 

```json
Lines executed:26.80% of 3186

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1960/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1960.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1960/Analysis_dynamic_valgrind-v3.13.0-B1960.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** b39f58ed27ff19b72bc09fd4c14d3080bedff773
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1961/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1961/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1961/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1961.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1961/Analysis_dynamic_valgrind-v3.13.0-B1961.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** b39f58ed27ff19b72bc09fd4c14d3080bedff773
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1962/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1962/ossec.log)
**Coverage:** 

```json
Lines executed:26.77% of 3186

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1962/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1962.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1962/Analysis_dynamic_valgrind-v3.13.0-B1962.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** b39f58ed27ff19b72bc09fd4c14d3080bedff773
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1965/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1965/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1965/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1965.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1965/Analysis_dynamic_valgrind-v3.13.0-B1965.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** b39f58ed27ff19b72bc09fd4c14d3080bedff773
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1966/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1966/ossec.log)
**Coverage:** 

```json
Lines executed:26.80% of 3186

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1966/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1966.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1966/Analysis_dynamic_valgrind-v3.13.0-B1966.log)

**Pull request merged**